cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -135,7 +136,28 @@ func createPod(ctx context.Context, events publisher, req *task.CreateTaskReques
 		host:   parent,
 	}
 	if oci.IsWCOW(s) {
-		// For WCOW we fake out the init task since we dont need it.
+		// For WCOW we fake out the init task since we dont need it. We only
+		// need to provision the guest network namespace.
+		nsid := ""
+		if s.Windows != nil && s.Windows.Network != nil {
+			nsid = s.Windows.Network.NetworkNamespace
+		}
+
+		// TODO: JTERRY75 - Do we need to support other than RS5 hosts here?
+		if nsid != "" {
+			endpoints, err := hcsoci.GetNamespaceEndpoints(nsid)
+			if err != nil {
+				return nil, err
+			}
+			err = parent.AddNetNS(nsid)
+			if err != nil {
+				return nil, err
+			}
+			err = parent.AddEndpointsToNS(nsid, endpoints)
+			if err != nil {
+				return nil, err
```

Comment:
Does this not need a cleanup on failure similar to how you have in `CreateContainer` in hcsoci/create.go? Should that code be common anyway as it's almost identical?

Reply:
Good call. Fixed.